Read hdf does not close store (#25863)
pandas-dev/pandas: doc/source/whatsnew/v0.25.0.rst

- Bug in :meth:`DataFrame.to_string` and :meth:`DataFrame.to_latex` that would lead to incorrect output when the ``header`` keyword is used (:issue:`16718`)
- Bug in :func:`read_csv` not properly interpreting the UTF8 encoded filenames on Windows on Python 3.6+ (:issue:`15086`)
- Improved performance in :meth:`pandas.read_stata` and :class:`pandas.io.stata.StataReader` when converting columns that have missing values (:issue:`25772`)

> - Bug in :func:`read_hdf` not properly close store when key is not found (:issue:`25766`)

Reviewer: ```suggestion
- Bug in :func:`read_hdf` not properly closing store after a KeyError is raised (:issue:`25766`)
```

> - Bug in :func:`read_hdf` not properly closing store after a KeyError is raised (:issue:`25766`)

Reviewer 2: use double backticks around ``KeyError``

Reviewer: Looks like this was done - going to merge
